src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5214,13 +5196,12 @@ private FieldDeclarationSyntax ParseConstantFieldDeclaration(SyntaxList<Attribut
             modifiers.Add(this.EatToken(SyntaxKind.ConstKeyword));
 
             var type = this.ParseType();
-
-            var variables = _pool.AllocateSeparated<VariableDeclaratorSyntax>();
-            this.ParseVariableDeclarators(type, VariableFlags.Const, variables, parentKind);
             return _syntaxFactory.FieldDeclaration(
                 attributes,
                 modifiers.ToList(),
-                _syntaxFactory.VariableDeclaration(type, _pool.ToListAndFree(variables)),
+                _syntaxFactory.VariableDeclaration(
+                    type,
+                    this.ParseFieldDeclarationVariableDeclarators(type, VariableFlags.Const, parentKind)),
```

Comment:
>ParseFieldDeclarationVariableDeclarators [](http://example.com/codeflow?start=25&length=40)

It looks like `ParseConstantFieldDeclaration()` is now indirectly saving and restoring `_termState` when parsing variable declarators. What is the impact of that change?

Reply:
honestly, it's likely better there so that parsing a constant field is the same as parsing a normal field.  TBH, it's kinda strange that we even have this method instead of just consuming 'const' in the caller and calling into normal field decl parsing. 

The only real difference here is that `;` is mixed into the terminator tokens for this construct.  but that's totally fine and desirable for constant fields.